packages/web/src/components/session-menu/index.tsx

```diff
@@ -0,0 +1,393 @@
+import { marked } from 'marked';
+import { cn } from '@/lib/utils';
+import { useHotkeys } from 'react-hotkeys-hook';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { SessionChannel } from '@/clients/websocket';
+import { ListIcon, PackageIcon, SettingsIcon, KeySquareIcon, XIcon } from 'lucide-react';
+import type { SessionType } from '@/types';
+import KeyboardShortcutsDialog from '@/components/keyboard-shortcuts-dialog';
+import FeedbackDialog from '@/components/feedback-dialog';
+import { Button } from '@/components/ui/button';
+import { Sheet, SheetContent } from '@/components/ui/sheet';
+// import {
+//   NavigationMenu,
+//   NavigationMenuContent,
+//   NavigationMenuList,
+//   NavigationMenuItem,
+//   NavigationMenuTrigger,
+// } from '@/components/ui/navigation-menu';
+import SessionMenuPanelTableOfContents from './table-of-contents-panel';
+import SessionMenuPanelPackages from './packages-panel';
+import SessionMenuPanelSettings from './settings-panel';
+// import SessionMenuPanelSecrets from './secrets-panel';
+
+export type SessionMenuPanelContentsProps = {
+  session: SessionType;
+  channel: SessionChannel;
+  openDepsInstallModal: () => void;
+};
+
+export const SESSION_MENU_PANELS = [
+  {
+    name: 'tableOfContents' as const,
+    icon: ListIcon,
+    openWidthInPx: 312,
+    contents: () => <SessionMenuPanelTableOfContents />,
+  },
+  {
+    name: 'packages' as const,
+    icon: PackageIcon,
+    openWidthInPx: 480,
+    contents: ({ session, openDepsInstallModal }: SessionMenuPanelContentsProps) => (
+      <SessionMenuPanelPackages session={session} openDepsInstallModal={openDepsInstallModal} />
+    ),
+  },
+  {
+    name: 'settings' as const,
+    icon: SettingsIcon,
+    openWidthInPx: 480,
+    contents: (props: SessionMenuPanelContentsProps) => <SessionMenuPanelSettings {...props} />,
+  },
+  // NOTE: re-enable this in the follow up change!
+  // {
+  //   name: 'secrets' as const,
+  //   icon: KeySquareIcon,
+  //   openWidthInPx: 480,
+  //   contents: () => <SessionMenuPanelSecrets />,
+  // },
+];
```

Comment:
This data structure controls rendering all the side panels. For now, the secrets panel is disabled.